Add blog post to reflect state of PodReadyToStartContainersCondition (#43932)
kubernetes/website: content/en/blog/_posts/2023-12-14-PodReadyToStartContainersCondition-in-beta.md

---
layout: blog

> title: "Kubernetes 1.29: PodReadyToStartContainersCondition in beta"

Reviewer: ```suggestion
title: "Kubernetes 1.29: PodReadyToStartContainersCondition Moves to Beta"
```

date: 2023-12-14
slug: pod-ready-to-start-containers-condition-now-in-beta
---

**Author**: zefeng chen, Kevin Hannon(Red Hat)

> With recent release of Kubernetes 1.29 the `PodReadyToStartContainers`[condition](/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions) is available by default.

Author: updated.

The kubelet manages the value for that condition throughout a Pod's lifecycle, 
in the status field of a pod. The kubelet will use the PodReadyToStartContainers 
condition to accurately surface the initialization state of a pod,
from the perspective of pod sandbox creation and network configuration by a container runtime.

> ## What's the motivation and why we need it?

Reviewer: ```suggestion
## What's the motivation for this feature?
```

The [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we) recommends avoiding the use of the word "we"

> Cluster administrators currently do not have a clear and easily accessible way to view the completion of pod's sandbox creation

Reviewer: The [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-that-will-soon-be-out-of-date) recommends avoiding the use of the word "currently". Can we rephrase this?

> Cluster administrators currently do not have a clear and easily accessible way to view the completion of pod's sandbox creation

Author: updated.

> and initialization. The current Initialized condition in pods tracks the execution of init containers. 

Reviewer: Also, instead of "current" here, can we refer to the the version for which this is true? e.g. "As of 1.28, the Initialized condition in Pods tracks..."

[style guide](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-that-will-soon-be-out-of-date)

This distinction is particularly important in multi-tenant clusters where tenants own the pod specifications, including the set of init containers, 
while cluster administrators manage storage plugins, networking plugins, and container runtime handlers. 
Therefore, there is a need for an improved mechanism to provide cluster administrators with a clear and 
comprehensive view of pod sandbox creation completion and container readiness.

## What's the benefit?

1.Improved Visibility: Cluster administrators gain a clearer and more comprehensive view of pod sandbox creation completion and container readiness. 
This enhanced visibility allows them to make better-informed decisions and troubleshoot issues more effectively.

2.Metric Collection and Monitoring: Monitoring services can leverage the fields associated with the PodReadyToStartContainers condition to 
report sandbox creation state and latency. 
Metrics can be collected at per-pod cardinality or aggregated based on various 
properties of the pod, such as volumes, storage class of Persistent Volume 
Claims (PVCs), runtime class, 
custom annotations for CNI and IPAM plugins, and arbitrary labels and annotations on pods. 

> This enables comprehensive monitoring and analysis of pod readiness across the cluster.

Author: updated.

> 3.Enhanced Troubleshooting: With a more accurate representation of pod sandbox creation and container readiness, 

Reviewer: ```suggestion
3.Enhanced Troubleshooting: With a more accurate representation of Pod sandbox creation and container readiness, 
```

cluster administrators can quickly identify and address any issues that may arise during the initialization process. 
This leads to improved troubleshooting capabilities and reduced downtime.

### What’s next?

Depending on feedback and adoption, the Kubernetes team promoted
the `PodReadyToStartContainersCondition` condition to Beta in 1.29.

> And we truly need your feedback on this feature!

Reviewer: This bit doesn't make sense to me. This *is* being promoted to beta right? So that bit doesn't _depend_ on feedback. Perhaps it should say something along the lines of:
```
Due to feedback and adoption, the Kubernetes team promoted `PodReadyToStartContainersCondition` to Beta in 1.29. Your comments will help determine if this condition continues forward to get promoted to GA, so please submit additional feedback on this feature!
```

> And we truly need your feedback on this feature!

Reviewer: ```suggestion
And SIG Node truly need your feedback on this feature!
```

### How can I learn more?

Please check out the
[documentation](/docs/concepts/workloads/pods/pod-lifecycle/) for the
`PodReadyToStartContainersCondition` to learn more about it and how it fits in relation to

> other pod conditions.

Reviewer: ```suggestion
other Pod conditions.
```
